vignettes/waiting_list_sim.Rmd

```diff
@@ -0,0 +1,460 @@
+
+current_wl1 <- wl_simulator(start_date = start_date,
+                            end_date = end_date,
+                            demand = 100,
+                            capacity = 80)
+
+tail(wl_queue_size(current_wl1))
+```
+
+
+As an alternative, we could manually dump 840 open pathways into the current waiting list the day before our start date for the main simulation.
+
+```{r dump_wl_setup}
+current_wl1 <-
+  data.frame(
+    Referral = rep(as.Date("2023-03-31"), 840)
+    , Removal = rep(as.Date(NA), 840)
+  )
+
+# Only a single day in our waiting list so we don't need `tail()` this time
+wl_queue_size(current_wl1)
+```
+
+## What is the target for this service?
+
+If we are governing our resources well, we want to allow a sustainable queue as this allows patients to be flexibly scheduled whilst still meeting our targets.  All capacity comes at a cost to the system, so we are seeking a balance between demand and capacity, with a sustainable waiting list where people are seen within a given target time.
+
+So our question is: what is the sustainable queue size for delivering 92% at 18 weeks, given 100 weekly demand?
+
+```{r target1}
+target_queue1 <-
+  calc_target_queue_size(demand = 100, target_wait = 18, factor = qt92)
+target_queue1
+```
+
+We have a target queue size of `r round(target_queue1, 2)`, and currently running a waiting list in excess of this, and load is >1, so the list will keep growing, without intervention, and average wait times will get longer.
+
+## Options for management
+
+There are two ways to bring a waiting list like this down:
+
+1. Raise the capacity ("more flowing out")
+2. Reduce the demand ("fewer flowing in")
+
+### 1. Raising capacity
+
+Let's imagine we are able to put an additional clinic in place that offers 25 more slots per week, taking our waiting list pressure down. We need to remember to include our waiting list into the simulation function, so it does not start from zero.  We will run the simulation for one year then plot the waiting list.
+
+```{r sim_raise_capacity}
+raised_capacity_wl <-
+  wl_simulator(start_date = as.Date("2023-04-01"),
+               end_date = as.Date("2024-03-31"),
+               demand = 100,
+               capacity = 105,
+               waiting_list = current_wl1)
+
+raised_capacity_queue <- wl_queue_size(raised_capacity_wl)
+
+tail(raised_capacity_queue)
+```
+
+### 2. Reducing demand
+
+Instead of raising capacity, let's imagine we have put in new referral guidance that reduces the number of referrals coming in to the system, resulting in 5 fewer referrals per week. Again, we need to remember to include our waiting list into the simulation function, so it does not start from zero.  We will run the simulation for one year then plot the waiting list.
+
+```{r sim_reduce_demand}
+reduced_demand_wl <-
+  wl_simulator(start_date = as.Date("2023-04-01"),
+               end_date = as.Date("2024-03-31"),
+               demand = 95,
+               capacity = 100,
```

Comment:
Capacity in the original example was 80. Should it be the same here? Ignore me if not...

Reply:
Very good spot!  Yes, you are right.  Will update that.